src/common/cockpitwebinject.c

```diff
@@ -84,62 +87,68 @@ cockpit_web_inject_push (CockpitWebFilter *filter,
   const gchar *mark, *data, *pos;
   gsize mark_len, data_len, at;
   GBytes *bytes;
+  gsize written;
 
   /* Stop searching once injected */
-  if (self->inject)
+  mark = g_bytes_get_data (self->marker, &mark_len);
+  data = g_bytes_get_data (block, &data_len);
+  written = at = 0;
```

Comment:
I think this whole loop would be cleaner if at were a gchar pointer instead of an index, but it's not a must.